browse/controllers/archive_page/__init__.py

```diff
@@ -0,0 +1,137 @@
+"""Archive landing page."""
+
+from datetime import date
+from typing import Dict, Any, Tuple, List
+
+from flask import Response, url_for
+
+from arxiv import status
+from arxiv.taxonomy.definitions import ARCHIVES, CATEGORIES, ARCHIVES_SUBSUMED
+
+from browse.controllers.archive_page.by_month_form import ByMonthForm
+from browse.controllers.archive_page.catchup_form import CatchupForm
+from browse.services.util.response_headers import abs_expires_header
+
+
+def get_archive(archive_id: str) -> Response:
+    """Gets archive page."""
+    data: Dict[str, Any] = {}
+    response_headers: Dict[str, Any] = {}
+
+    if archive_id == "list":
+        return archive_index(archive_id, status=status.HTTP_200_OK)
+
+    archive = ARCHIVES.get(archive_id, None)
+    if not archive:
+        archive_id = CATEGORIES.get(archive_id, {}).get("in_archive", None)
+        archive = ARCHIVES.get(archive_id, None)
+        if not archive:
+            return archive_index(archive_id, status=status.HTTP_404_NOT_FOUND)
```

Comment:
It might be nice to include a message with this 404 at the top of the archive index, per classic.

`No archive 'blah' -- available archives are:`